k8s/network.go

```diff
@@ -232,6 +244,20 @@ func (a *networkImpl) Stop(ctx context.Context) error {
 // until it is reachable.
 // Assumes [a.nodesLock] isn't held.
 func (a *networkImpl) AddNode(cfg node.Config) (node.Node, error) {
+	a.nodesLock.RLock()
+	isStopped := a.isStopped()
+	a.nodesLock.RUnlock()
+
+	// TODO fix this is race condition:
+	// Stop() can be called after the lock is released above.
+	// If that happens, we'll create the pod inside launchNodes
+	// and then never destroy it.
+	// launchNodes assumes [a.nodesLock] isn't held so we can't just
+	// hold the lock inside this method to fix this race condition.
```

Comment:
`launchNodes` could have a `select` with a `closedOnStop` case?

Reply:
Fabio and I talked about this and we'll patch in a follow-up PR